<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from train_robot.urdf.xacro         | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="train" xmlns:xacro="http://ros.org/wiki/xacro">
  <link name="base_footprint"/>
  <!--
       Base link is set at the bottom of the base mould.
       This is done to be compatible with the way base link
       was configured for turtlebot 1. Refer to
       
       https://github.com/turtlebot/turtlebot/issues/40
       
       To put the base link at the more oft used wheel
       axis, set the z-distance from the base_footprint
       to 0.352.
      -->
  <joint name="base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.22"/>
    <parent link="base_footprint"/>
    <child link="base_link"/>
  </joint>
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.4 0.3 0.05"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
  </link>
  <link name="front_axle">
    <visual>
      <geometry>
        <cylinder length="0.3" radius="0.01"/>
      </geometry>
      <origin rpy="1.57075 0 0" xyz="0.0 0.0 0.0"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_front_axle" type="revolute">
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-6.28318" upper="6.28318" velocity="0.5"/>
    <origin rpy="0 -0.75 0" xyz="0.1275 0 0"/>
    <parent link="base_link"/>
    <child link="front_axle"/>
  </joint>
  <link name="rear_axle">
    <visual>
      <geometry>
        <cylinder length="0.3" radius="0.01"/>
      </geometry>
      <origin rpy="1.57075 0 0" xyz="0.0 0.0 0.0"/>
      <material name="white"/>
    </visual>
  </link>
  <joint name="base_to_rear_axle" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-6.28318" upper="6.28318" velocity="0.5"/>
    <origin rpy="0 0.75 0" xyz="-0.136 0 0"/>
    <parent link="base_link"/>
    <child link="rear_axle"/>
  </joint>
  <link name="front_right_leg">
    <visual>
      <geometry>
        <box size="0.05 .05 0.117"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0585"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_front_right_leg" type="fixed">
    <axis xyz="1 0 0"/>
    <origin rpy="0 3.1415 1.57075" xyz="0 0.175 0"/>
    <parent link="front_axle"/>
    <child link="front_right_leg"/>
  </joint>
  <link name="front_right_lowleg">
    <visual>
      <geometry>
        <box size="0.04 .05 0.041"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0205"/>
      <material name="white"/>
    </visual>
  </link>
  <joint name="leg_to_front_right_lowleg" type="continuous">
    <axis xyz="-1 0 0"/>
    <limit effort="1000.0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="-0.04 0.0 0.117"/>
    <parent link="front_right_leg"/>
    <child link="front_right_lowleg"/>
    <mimic joint="base_to_front_axle" multiplier="1" offset="0.75"/>
  </joint>
  <link name="front_right_turnleg">
    <visual>
      <geometry>
        <box size="0.02 .05 .05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.025"/>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="leg_to_front_right_turnleg" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.1415" upper="3.1415" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.041"/>
    <parent link="front_right_lowleg"/>
    <child link="front_right_turnleg"/>
  </joint>
  <link name="front_right_wheel">
    <visual>
      <geometry>
        <mesh filename="package://train/urdf/meshes/hjul.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="front_right_wheel_joint" type="continuous">
    <axis xyz="0 0 1"/>
    <parent link="front_right_turnleg"/>
    <child link="front_right_wheel"/>
    <origin rpy="0 1.57075 0" xyz="0 0 0.05"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
  <link name="front_left_leg">
    <visual>
      <geometry>
        <box size="0.05 .05 0.117"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0585"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_front_left_leg" type="fixed">
    <axis xyz="1 0 0"/>
    <origin rpy="0 3.1415 1.57075" xyz="0 -0.175 0"/>
    <parent link="front_axle"/>
    <child link="front_left_leg"/>
  </joint>
  <link name="front_left_lowleg">
    <visual>
      <geometry>
        <box size="0.04 .05 0.041"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0205"/>
      <material name="white"/>
    </visual>
  </link>
  <joint name="leg_to_front_left_lowleg" type="continuous">
    <axis xyz="-1 0 0"/>
    <limit effort="1000.0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0.04 0.0 0.117"/>
    <parent link="front_left_leg"/>
    <child link="front_left_lowleg"/>
    <mimic joint="base_to_front_axle" multiplier="1" offset="0.75"/>
  </joint>
  <link name="front_left_turnleg">
    <visual>
      <geometry>
        <box size="0.02 .05 .05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.025"/>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="leg_to_front_left_turnleg" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.1415" upper="3.1415" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.041"/>
    <parent link="front_left_lowleg"/>
    <child link="front_left_turnleg"/>
  </joint>
  <link name="front_left_wheel">
    <visual>
      <geometry>
        <mesh filename="package://train/urdf/meshes/hjul.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="front_left_wheel_joint" type="continuous">
    <axis xyz="0 0 1"/>
    <parent link="front_left_turnleg"/>
    <child link="front_left_wheel"/>
    <origin rpy="0 1.57075 0" xyz="0 0 0.05"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
  <link name="rear_right_leg">
    <visual>
      <geometry>
        <box size="0.05 .05 0.117"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0585"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_rear_right_leg" type="fixed">
    <axis xyz="1 0 0"/>
    <origin rpy="0 3.1415 1.57075" xyz="0 0.175 0"/>
    <parent link="rear_axle"/>
    <child link="rear_right_leg"/>
  </joint>
  <link name="rear_right_lowleg">
    <visual>
      <geometry>
        <box size="0.04 .05 0.041"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0205"/>
      <material name="white"/>
    </visual>
  </link>
  <joint name="leg_to_rear_right_lowleg" type="continuous">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="-0.04 0.0 0.117"/>
    <parent link="rear_right_leg"/>
    <child link="rear_right_lowleg"/>
    <mimic joint="base_to_rear_axle" multiplier="1" offset="0.75"/>
  </joint>
  <link name="rear_right_turnleg">
    <visual>
      <geometry>
        <box size="0.02 .05 .05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.025"/>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="leg_to_rear_right_turnleg" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.1415" upper="3.1415" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.041"/>
    <parent link="rear_right_lowleg"/>
    <child link="rear_right_turnleg"/>
  </joint>
  <link name="rear_right_wheel">
    <visual>
      <geometry>
        <mesh filename="package://train/urdf/meshes/hjul.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="rear_right_wheel_joint" type="continuous">
    <axis xyz="0 0 1"/>
    <parent link="rear_right_turnleg"/>
    <child link="rear_right_wheel"/>
    <origin rpy="0 1.57075 0" xyz="0 0 0.05"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
  <link name="rear_left_leg">
    <visual>
      <geometry>
        <box size="0.05 .05 0.117"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0585"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_rear_left_leg" type="fixed">
    <axis xyz="1 0 0"/>
    <origin rpy="0 3.1415 1.57075" xyz="0 -0.175 0"/>
    <parent link="rear_axle"/>
    <child link="rear_left_leg"/>
  </joint>
  <link name="rear_left_lowleg">
    <visual>
      <geometry>
        <box size="0.04 .05 0.041"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0205"/>
      <material name="white"/>
    </visual>
  </link>
  <joint name="leg_to_rear_left_lowleg" type="continuous">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0.04 0.0 0.117"/>
    <parent link="rear_left_leg"/>
    <child link="rear_left_lowleg"/>
    <mimic joint="base_to_rear_axle" multiplier="1" offset="0.75"/>
  </joint>
  <link name="rear_left_turnleg">
    <visual>
      <geometry>
        <box size="0.02 .05 .05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.025"/>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="leg_to_rear_left_turnleg" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.1415" upper="3.1415" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.041"/>
    <parent link="rear_left_lowleg"/>
    <child link="rear_left_turnleg"/>
  </joint>
  <link name="rear_left_wheel">
    <visual>
      <geometry>
        <mesh filename="package://train/urdf/meshes/hjul.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="rear_left_wheel_joint" type="continuous">
    <axis xyz="0 0 1"/>
    <parent link="rear_left_turnleg"/>
    <child link="rear_left_wheel"/>
    <origin rpy="0 1.57075 0" xyz="0 0 0.05"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
</robot>

